File: StarWarsTcgApi/StarWarsTcgApi.Domain/Models/GameStatistics.cs
using System.ComponentModel.DataAnnotations;

namespace StarWarsTcgApi.Domain.Models
{
    public class GameStatistics
    {
        [Required]
        public string UserId { get; set; } = string.Empty;
        public string AvatarUrl { get; set; } = string.Empty; //TODO: AvatarId?
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;

        public int TotalGames { get; set; }
        public int LightGames { get; set; }
        public int LightWins { get; set; }
        public int DarkGames { get; set; }
        public int DarkWins { get; set; }
        public int TotalWins { get; set; }
    }
}

File: StarWarsTcg.Web/src/app/models/GameStatistics.ts

export class GameStatistics {

    public userId: string;
    public avatarUrl: string; //TODO: Fix this up to match the User.AvatarId + lookup
    public firstName: string;
    public lastName: string;

    public totalGames: number;
    public lightGames: number;
    public lightWins: number;
    public darkGames: number;
    public darkWins: number;
    public totalWins: number;

    /**
     * Constructor for the GameStatistics class. (vwGameStatistics)
     * Initializes all properties with default values or provided ones.
     * @param initialData An optional object to initialize the properties.
     */
    constructor(initialData?: Partial<GameStatistics>) {
        this.userId = initialData?.userId ?? '';
        this.avatarUrl = initialData?.avatarUrl ?? '';
        this.firstName = initialData?.firstName ?? '';
        this.lastName = initialData?.lastName ?? '';
        
        this.totalGames = initialData?.totalGames ?? 0;
        this.lightGames = initialData?.lightGames ?? 0;
        this.lightWins = initialData?.lightWins ?? 0;
        this.darkGames = initialData?.darkGames ?? 0;
        this.darkWins = initialData?.darkWins ?? 0;
        this.totalWins = initialData?.totalWins ?? 0;
    }

}
